src/ngraph/pass/constant_folding_non_zero.cpp

```diff
@@ -18,169 +18,59 @@
 
 #include "constant_folding.hpp"
 #include "ngraph/op/non_zero.hpp"
+#include "ngraph/runtime/reference/non_zero.hpp"
```

Comment:
This entire `constant_folding_non_zero.cpp` should be deleted.

Reply:
I can't delete this constant_folding_non_zero.cpp as in this PR, the deletion will happen in my working PR with adding dynamic evaluate. This PR is just for adding the reference kernel implementation, and I am using the constant folding as a way to test the reference kernel implementation before the dynamic evaluate PR is ready.